terraform/aws.tf

```diff
@@ -502,3 +487,237 @@ resource "aws_iam_role_policy_attachment" "Cloudguard_quicksight_s3_access_right
   role       = aws_iam_role.quicksight_service_role.name
   policy_arn = resource.aws_iam_policy.cloudguard_quicksight_s3_policy.arn
 }
+
+################
+## SSO Config ##
+################
+
+# Quicksight reader access role
+resource "aws_iam_role" "quicksight_reader_role" {
+  name                 = "BCGOV_CORE_Quicksight_reader"
+  max_session_duration = 21600
+
+  assume_role_policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Federated": "arn:aws:iam::${data.aws_caller_identity.current.account_id}:saml-provider/BCGovKeyCloak-tmhl5tvs"
```

Comment:
`BCGovKeyCloak-tmhl5tvs` should be an input variable. This changes if keycloak is in a different realm.

Reply:
Yes I've changed it to var.kc_realm